xtask/src/commands/generate_rom_symbols.rs

```diff
@@ -0,0 +1,113 @@
+use std::{
+    collections::HashSet,
+    fs,
+    fs::OpenOptions,
+    io::Write,
+    path::{Path, PathBuf},
+};
+
+use anyhow::{Context, Error};
+use regex::Regex;
+
+use crate::Chip;
+
+/// Generates or updates the `generated_rom_symbols.rs` file with ROM symbol markers
+/// for the specified target chip.
+pub(crate) fn generate_rom_symbols(workspace: &Path, target_chip: &Chip) -> Result<(), Error> {
+    // Convert chip to lowercase string for paths and modules
+    let chip_name = target_chip.to_string();
+
+    let rom_ld_dir = workspace.join(format!("ld/{}/rom", chip_name));
+
+    let dest_path = workspace.join("src/generated_rom_symbols.rs");
+
+    let current_chip_module_content =
+        generate_module_content(&rom_ld_dir, &chip_name, &target_chip)?;
+
+    let mut out_file = OpenOptions::new()
+        .create(true)
+        .write(true)
+        .truncate(true)
+        .open(&dest_path)
+        .context(format!(
+            "Failed to open {} for writing",
+            dest_path.display()
+        ))?;
+
+    writeln!(out_file, "{}", current_chip_module_content).context(format!(
+        "Failed to write module content for chip {} to generated_rom_symbols.rs",
+        chip_name
+    ))?;
+
+    Ok(())
+}
+
+// Helper function to handle the parsing and content generation.
+fn generate_module_content(
+    rom_ld_dir: &Path,
+    chip_name: &str,
+    _target_chip: &Chip,
+) -> Result<String, Error> {
+    let mut content = String::new();
+
+    let linker_scripts: Vec<PathBuf> = fs::read_dir(rom_ld_dir)?
+        .filter_map(|entry| {
+            let path = entry.ok()?.path();
+            if path.extension().and_then(|s| s.to_str()) == Some("ld") {
+                Some(path)
+            } else {
+                None
+            }
+        })
+        .collect();
+
+    content.push_str(&format!(
+            "#[cfg(all(feature = \"{chip}\", feature = \"__internal_rom_symbols\"))]\npub mod rom_symbols {{\n",
```

Comment:
inclusion of the module is already gated by `__internal_rom_symbols` and you ditched the idea of generating the code for multiple chips at the same time so the cfg seems to be unnecessary?